crates/omnix-health/src/check/flake_enabled.rs

```diff
@@ -15,22 +15,114 @@ impl Checkable for FlakeEnabled {
+            NixInstallationType::DeterminateSystems => (
+                true, // Determinate Systems Nix has flakes enabled by default
+                "Flakes enabled via Determinate Systems Nix".to_string(),
+                CheckResult::Green,
+            ),
+            NixInstallationType::Official => {
+                let flakes_enabled =
+                    val.contains(&"flakes".to_string()) && val.contains(&"nix-command".to_string());
+                let info_msg = format!("experimental-features = {}", val.join(" "));
+                let result = match flakes_enabled {
+                    true => CheckResult::Green,
+                    false => CheckResult::Red {
+                        msg: "Nix flakes are not enabled".into(),
+                        suggestion: "See https://nixos.wiki/wiki/Flakes#Enable_flakes".into(),
+                    },
+                };
+                (flakes_enabled, info_msg, result)
+            }
+        };
+
         let check = Check {
             title: "Flakes Enabled".to_string(),
-            info: format!("experimental-features = {}", val.join(" ")),
-            result: if val.contains(&"flakes".to_string())
-                && val.contains(&"nix-command".to_string())
-            {
-                CheckResult::Green
-            } else {
-                CheckResult::Red {
-                    msg: "Nix flakes are not enabled".into(),
-                    suggestion: "See https://nixos.wiki/wiki/Flakes#Enable_flakes".into(),
-                }
-            },
+            info: info_msg,
+            result,
             required: true,
         };
 
         vec![("flake-enabled", check)]
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::traits::CheckResult;
+    use nix_rs::{config::NixConfig, info::NixInfo, version::NixVersion};
+
+    fn mock_nix_config(experimental_features: &str) -> NixConfig {
+        // Mock minimal nix.conf: experimental-features = flakes nix-command
+        let features_array = experimental_features
+            .split_whitespace()
+            .map(|s| format!("\"{}\"", s))
+            .collect::<Vec<_>>()
+            .join(",");
+        let config_json = format!(
+            r#"{{
+            "experimental-features": {{ "value": [{}], "defaultValue": [], "description": "" }},
+            "cores": {{ "value": 1, "defaultValue": 1, "description": "" }},
+            "extra-platforms": {{ "value": [], "defaultValue": [], "description": "" }},
+            "flake-registry": {{ "value": "", "defaultValue": "", "description": "" }},
+            "max-jobs": {{ "value": 1, "defaultValue": 1, "description": "" }},
+            "substituters": {{ "value": [], "defaultValue": [], "description": "" }},
+            "system": {{ "value": "x86_64-linux", "defaultValue": "x86_64-linux", "description": "" }},
+            "trusted-users": {{ "value": [], "defaultValue": [], "description": "" }}
+        }}"#,
```

Comment:
[Just](https://github.com/juspay/omnix/pull/473#discussion_r2283433150) mock the `nix.conf` file itself (keep it 1 line; for experimental-features), and then parse it.

`nix.conf` is in INI format, not JSON, as you are mocking here.